.github/shared/src/array.js

```diff
@@ -11,6 +11,17 @@ export async function filterAsync(array, asyncPredicate) {
   return array.filter((_, i) => results[i]);
 }
 
+/**
+ * @template T,U
+ * @param {T[]} array
+ * @param {(item: T, index: number, array: T[]) => Promise<U[]>} asyncMapper
+ * @returns {Promise<U[]>}
+ */
+export async function flatMapAsync(array, asyncMapper) {
```

Comment:
I ended up not using this, but we'll need it soon enough, so let's just add it now